plugins/optimization-detective/webpack.config.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+const WebpackBar = require( 'webpackbar' );
+const CopyWebpackPlugin = require( 'copy-webpack-plugin' );
+
+/**
+ * Internal dependencies
+ */
+const {
+	getPluginRootPath,
+	assetDataTransformer,
+} = require( '../../bin/webpack/utils' );
+
+/**
+ * WordPress dependencies
+ */
+const defaultConfig = require( '@wordpress/scripts/config/webpack.config' );
+
+const sharedConfig = {
+	...defaultConfig,
+	entry: {},
+	output: {},
+};
+
+/**
+ * Webpack Config: Optimization Detective
+ *
+ * @param {*} env Webpack environment
+ * @return {Object} Webpack configuration
+ */
+const optimizationDetective = ( env ) => {
+	if ( env.plugin && env.plugin !== 'optimization-detective' ) {
+		return {
+			entry: {},
+			output: {},
+		};
+	}
+
+	const source = path.resolve(
+		getPluginRootPath(),
+		'node_modules/web-vitals'
+	);
+	const destination = path.resolve(
+		getPluginRootPath(),
+		'plugins/optimization-detective/detection'
+	);
+
+	return {
+		...sharedConfig,
+		name: 'optimization-detective',
+		plugins: [
+			new CopyWebpackPlugin( {
+				patterns: [
+					{
+						from: `${ source }/dist/web-vitals.js`,
+						to: `${ destination }/web-vitals.js`,
+					},
+					{
+						from: `${ source }/package.json`,
+						to: `${ destination }/web-vitals.asset.php`,
+						transform: {
+							transformer: assetDataTransformer,
+							cache: false,
+						},
+					},
+				],
+			} ),
+			new WebpackBar( {
+				name: 'Building Optimization Detective Assets',
+				color: '#2196f3',
+			} ),
+		],
+	};
+};
+
+module.exports = optimizationDetective;
```

Comment:
Which parts of this file are actually needed?

Background is I don't recall that we discussed having individual webpack scripts per plugin before, and since the only plugin that needs a build process to function at all so far is this one, I think we should go with a more pragmatic solution. Why not include this Optimization Detective specific web-vitals step in the main `webpack.config.js`? In other words, the way this was before.

We can consider optimizing it later, let's just get to an MVP solution now. Decisions that affect the build infrastructure of the entire repo shouldn't be made in a plugin's feature branch PR.

Reply:
Per https://github.com/WordPress/performance/pull/1080#issuecomment-2014734993 it seems like you're saying 767c895ef46636b4af1b90bdafecca8e1211fa0d should be reverted?

Reply:
@westonruter Yes, I think that's a much simpler solution for now, especially since this is the only plugin that needs a build process.

Reply:
Reverted in 0e07d110